resourcemanager/recaser/recaser_test.go

```diff
@@ -0,0 +1,129 @@
+package recaser_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/recaser"
+)
+
+func TestRecaserWithIncorrectCasing(t *testing.T) {
```

Comment:
could we add a couple of the awkward test cases:

1. There's a handful of APIs that return the full URI with a port in some cases e.g. `https://management.azure.com:80/subscriptions/12345`
2. Could we add a Data Plane URI here to ensure this is ignored too: `https://example.blob.storage.azure.com/container1` should be fine

As such could we add a couple of test cases covering those?